dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql

```diff
@@ -796,6 +797,7 @@ CREATE TABLE t_ds_task_instance (
 ) ;
 
 create index idx_task_instance_code_version on t_ds_task_instance (task_code, task_definition_version);
+create index idx_task_instance_process_instance_flag on t_ds_task_instance (process_instance_id, flag);
```

Comment:
Maybe we should also add to specific update directory

Reply:
I just use postgresql

and I found there are some foreign key in mysql
I think it could be removed to improve

Reply:
> I just use postgresql
> 
> and I found there are some foreign key in mysql I think it could be removed to improve

But as a PR, you should also add your change to other type of database, such as mysql and H2 to keep synchronous.
Also, you should add the index add in https://github.com/apache/dolphinscheduler/blob/489e7fe4e2d980031c91eac2a4823b9f1713659a/dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/postgresql/dolphinscheduler_ddl.sql and https://github.com/apache/dolphinscheduler/blob/489e7fe4e2d980031c91eac2a4823b9f1713659a/dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql

Reply:
@darkz1984 you did a good job, 

can you add these changes to mysql and H2 to keep the db file sync, I think it's not difficult for you ^_^

Reply:
> @darkz1984 you did a good job,
> 
> can you add these changes to mysql and H2 to keep the db file sync, I think it's not difficult for you ^_^

I am not use mysql or H2 with dolphinscheduler on my server, So, I think I could do this...

Reply:
And add an index for a big t_ds_task_instance with much rows online when dolphinscheduler is running in PostgreSQL is very easy and just cost few milliseconds. 
But this action in MySQL is very danger, It may lead table lock and long time cost,  My be an production accident.

Reply:
> @darkz1984 you did a good job,
> 
> can you add these changes to mysql and H2 to keep the db file sync, I think it's not difficult for you ^_^

I had try mysql :
https://github.com/apache/dolphinscheduler/pull/13080

Reply:
https://github.com/apache/dolphinscheduler/pull/13081
the post upgrade ddl

Reply:
@davidzollo

Reply:
And I think H2 database has no index, so, no need to add index for it.